tika-core/src/main/java/org/apache/tika/detect/CompositeEncodingDetector.java

```diff
@@ -71,10 +80,15 @@ public Charset detect(InputStream input, Metadata metadata) throws IOException {
     }
 
     public List<EncodingDetector> getDetectors() {
-        return Collections.unmodifiableList(detectors);
+        List<EncodingDetector> allDetectors = new LinkedList<>();
+        if (loader != null) {
+            allDetectors.addAll(loader.loadDynamicServiceProviders(EncodingDetector.class));
```

Comment:
Why don't you filter the dynamic loaded services?

Reply:
that part I now removed to make it similar to the structure of the DefatultDetector